### DISCUSSION TITLE
Timeout and frequent connection error. Anyone have a fix?

### DISCUSSION THREAD

**Discussion by simple1976** [kellerza/sunsynk#245] [General]
Hi brain thrust, anyone have a recommendation to address below issue.  The plugin requires restart every few days as it becomes unresponsive.   Also, get ongoing error as below.
Setup is 2 x DEYE / SUN-8K-SG05LP1-AU. Single phase, parallel setup. SolarMan dongle. 2023 model.

22:00:21 INFO    Queue update load_limit=Essentials
22:00:22 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
22:00:31 INFO    Queue update battery_max_discharge_current=180
22:00:32 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
22:03:31 ERROR   Error retry attempt(0): Connection already closed
22:05:00 ERROR   (1 in 5 min) OSError in read_sunsynk: timeout reading 10 registers from 210 poll_need_to_read
22:05:16 ERROR   Error retry attempt(0): Connection already closed
22:07:32 ERROR   Error retry attempt(0): Connection already closed
22:10:00 ERROR   (1 in 5 min) OSError in read_sunsynk2: timeout reading 1 registers from 219; timeout reading 6 registers from 243 poll_need_to_read
22:10:00 ERROR   (1 in 5 min) OSError in read_sunsynk2: timeout reading 1 registers from 219 poll_need_to_read
22:26:13 ERROR   Error retry attempt(0): Connection already closed
22:30:00 ERROR   (1 in 5 min) OSError in read_sunsynk: timeout reading 10 registers from 210 poll_need_to_read
22:58:32 ERROR   Error retry attempt(0): Connection already closed
23:00:00 ERROR   (1 in 5 min) OSError in read_sunsynk2: timeout reading 1 registers from 59 poll_need_to_read
23:00:00 INFO    Queue update prog6_charge=No Grid or Gen
23:00:01 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:00:10 INFO    Queue update prog6_capacity=5
23:00:11 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:00:20 INFO    Queue update load_limit=Essentials
23:00:22 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:00:30 INFO    Queue update battery_max_discharge_current=180
23:00:31 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
23:10:08 INFO    Queue update prog6_charge=No Grid or Gen
23:10:09 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:10:18 INFO    Queue update prog6_capacity=5
23:10:19 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:10:28 INFO    Queue update load_limit=Essentials
23:10:29 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:10:38 INFO    Queue update battery_max_discharge_current=180
23:10:39 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
23:20:09 INFO    Queue update prog6_charge=No Grid or Gen
23:20:10 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:20:19 INFO    Queue update prog6_capacity=5
23:20:21 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:20:29 INFO    Queue update load_limit=Essentials
23:20:31 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:20:39 INFO    Queue update battery_max_discharge_current=180
23:20:40 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
23:30:08 INFO    Queue update prog6_charge=No Grid or Gen
23:30:09 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:30:18 INFO    Queue update prog6_capacity=5
23:30:19 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:30:28 INFO    Queue update load_limit=Essentials
23:30:29 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:30:38 INFO    Queue update battery_max_discharge_current=180
23:30:39 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
23:34:17 ERROR   Error retry attempt(0): Connection already closed
23:35:00 ERROR   (1 in 5 min) OSError in read_sunsynk2: timeout reading 1 registers from 59 poll_need_to_read
23:40:08 INFO    Queue update prog6_charge=No Grid or Gen
23:40:09 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:40:18 INFO    Queue update prog6_capacity=5
23:40:19 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:40:28 INFO    Queue update load_limit=Essentials
23:40:29 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:40:38 INFO    Queue update battery_max_discharge_current=180
23:40:39 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
23:50:09 INFO    Queue update prog6_charge=No Grid or Gen
23:50:10 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
23:50:19 INFO    Queue update prog6_capacity=5
23:50:20 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
23:50:29 INFO    Queue update load_limit=Essentials
23:50:31 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
23:50:39 INFO    Queue update battery_max_discharge_current=180
23:50:40 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
00:00:00 INFO    Queue update prog6_charge=No Grid or Gen
00:00:01 INFO    Writing sensor prog6_charge=No Grid or Gen Registers:{279=0x0000} [Register 0-->0]
00:00:10 INFO    Queue update prog6_capacity=5
00:00:11 INFO    Writing sensor prog6_capacity=5 Registers:{273=0x0005} 
00:00:20 INFO    Queue update load_limit=Essentials
00:00:21 INFO    Writing sensor load_limit=Essentials Registers:{244=0x0001} 
00:00:30 INFO    Queue update battery_max_discharge_current=180
00:00:31 INFO    Writing sensor battery_max_discharge_current=180 Registers:{211=0x00b4} 
00:20:48 ERROR   Error retry attempt(0): Connection already closed
00:25:00 ERROR   (1 in 5 min) OSError in read_sunsynk: timeout reading 1 registers from 59 poll_need_to_read

**Comment by kellerza** [comment]
See #184
